test/race_cond_test.go

```diff
@@ -0,0 +1,227 @@
+		PrimaryDSName: testData.StepPrimaryDSName,
+		PrimaryDSType: "test",
+		CreateBy:      "WMAgent",
+		CreationDate:  1501192514, // Replace with fixed time
+	}
+
+	dataset := dbs.Dataset{
+		PhysicsGroupName:     testData.PhysicsGroupName,
+		ProcessedDSName:      testData.ProcDataset,
+		DataTierName:         testData.Tier,
+		DatasetAccessType:    testData.DatasetAccessType2,
+		Dataset:              testData.StepchainDataset,
+		PrepID:               "TestPrepID",
+		CreateBy:             "WMAgent",
+		LastModifiedBy:       "WMAgent",
+		CreationDate:         1501192514,
+		LastModificationDate: 1501192514,
+	}
+
+	processingEra := dbs.ProcessingEra{
+		ProcessingVersion: testData.ProcessingVersion,
+		CreateBy:          "WMAgent",
+		CreationDate:      1501192514,
+	}
+
+	acqEra := dbs.AcquisitionEra{
+		AcquisitionEraName: testData.AcquisitionEra,
+		StartDate:          123456789,
+		CreationDate:       1501192514,
+	}
+
+	fileCount := 1000
+
+	block := dbs.Block{
+		BlockName:      fmt.Sprintf("%s%d", testData.StepchainBlock, factor),
+		OriginSiteName: testData.Site,
+		FileCount:      int64(fileCount),
+		BlockSize:      20122119010,
+		CreationDate:   1501192514,
+	}
+
+	bulkBlock.DatasetConfigList = append(bulkBlock.DatasetConfigList, algo)
+	bulkBlock.PrimaryDataset = primDS
+	bulkBlock.Dataset = dataset
+	bulkBlock.ProcessingEra = processingEra
+	bulkBlock.AcquisitionEra = acqEra
+	// bulkBlock.DatasetParentList = []string{testData.ParentStepchainDataset}
+	bulkBlock.Block = block
+
+	for i := 0; i < fileCount; i++ {
+		f, fileConfig := createFile2(t, i*factor, factor)
+		bulkBlock.Files = append(bulkBlock.Files, f)
+		bulkBlock.FileConfigList = append(bulkBlock.FileConfigList, fileConfig)
+	}
+
+	return bulkBlock
+}
+
+// submit a bulkblock to DBSWriter in a goroutine
+func submitBulkBlock(t *testing.T, wg *sync.WaitGroup, errs chan<- string, tries int, block dbs.BulkBlocks) {
+	retries := tries
+	defer wg.Done()
+	data, err := json.Marshal(block)
+	if err != nil {
+		errs <- err.Error()
+	}
+	reader := bytes.NewReader(data)
+	ht := http.DefaultTransport.(*http.Transport).Clone()
+	ht.CloseIdleConnections()
+	ht.DisableKeepAlives = true
+	c := http.Client{Transport: ht}
+
+	resp, err := c.Post("http://localhost:8990/dbs-one-writer/bulkblocks", "application/json", reader)
+	if err != nil {
+		errs <- err.Error()
+	}
+	defer resp.Body.Close()
+	t.Logf("Try #%d, Block: %s, response: %d\n", retries, block.Block.BlockName, resp.StatusCode)
+	if resp.StatusCode != http.StatusOK {
+		if retries < 3 {
+			wg.Add(1)
+			retries++
+			t.Logf("Re-Trying for Block: %s", block.Block.BlockName)
+			time.Sleep(10 * time.Second)
+			go submitBulkBlock(t, wg, errs, retries, block)
+		} else {
+			msg := fmt.Sprintf("Submit bulkblocks failed on try #%d for block %s", retries, block.Block.BlockName)
+			errs <- msg
+		}
+	}
+
+	//return resp.StatusCode
+}
+
+// TestRaceConditions tests for the racing conditions in the DBSWriter bulkblocks API
```

Comment:
could you please add more description about what racing conditions we are talking here. Please add a link to our github ticket.